kart/tabular/working_copy/base.py

```diff
@@ -454,13 +454,43 @@ def diff_repo_to_working_copy(
             repo_diff.prune()
             return repo_diff
 
+    def diff_table_repo_to_working_copy(self, ds_diff):
+        """
+        Generates a diff between a working copy DB and the underlying repository tree,
+        for a single dataset.
+        """
+        with self.session():
+            repo_diff = RepoDiff()
+            repo_diff[self.path] = ds_diff
+        repo_diff.prune()
+        return repo_diff
+
     def _is_dataset_supported(self, dataset):
         """
         Returns False if the given dataset cannot be created in this working copy.
         Generally returns True since we do our best to create all datasets, even if not 100% accurately.
         """
         return True
 
+    def diff_table(self, table, meta_items):
+        """
+        Generates a diff between a working copy DB and the underlying repository tree,
+        for a single table in the given repository structure..
+        """
```

Comment:
this function actually doesn't look at any existing dataset at all - it generates a diff with only a 'new' side and no old side. I'd suggest renaming it to `get_diff_for_table_creation` or something similar, and updating the docstring to match